### PR TITLE
 [ios, build] Replace -symbols with -stripped build flavor

### LINKED ISSUE #12947: Dynamic framework is too big
Since I like reproducible builds, zero-config checkouts and ease of development (also when offline) among other reasons, I add the `Pods/` directory to github. It seems some build configuration has changed between the 0.9.2 and 0.9.3 pod build settings? Or do I need to tweak something on my end?

```
remote: Resolving deltas: 100% (36/36), completed with 33 local objects.
remote: warning: File Pods/Mapbox-iOS-SDK/dynamic/Mapbox.framework/Mapbox is 70.95 MB; this is larger than GitHub's recommended maximum file size of 50.00 MB
remote: error: GH001: Large files detected. You may want to try Git Large File Storage - https://git-lfs.github.com.
remote: error: Trace: 4260c1065b630e7cdf5f3974bb9fe8ca
remote: error: See http://git.io/iEPt8g for more information.
remote: error: File Pods/Mapbox-iOS-SDK/dynamic/Mapbox.framework.dSYM/Contents/Resources/DWARF/Mapbox is 247.02 MB; this exceeds GitHub's file size limit of 100.00 MB
To github.com:foo/bar.git
 ! [remote rejected]   develop -> develop (pre-receive hook declined)
```

It looks like it is built with DWARF debug symbols, is there an option to get the binary without that?

### Steps to reproduce

 1. install Apollo v0.9.2, update to 0.9.3
 2. observe repo bloat
 

### Configuration

**Mapbox SDK versions:** 
**iOS/macOS versions:** 
**Device/simulator models:**
**Xcode version:**

### LINKED ISSUE #13395: Deprecate -symbols builds & add new -stripped builds
~In order to fix long-standing symbolication issues (#8463) and~ to support developers that need smaller files to store in source control (#12947), we should:

1. Make the `-symbols` build flavor the new default build that we distribute via primary channels. (This is effectively a renaming of `Mapbox-iOS-SDK-symbols` to `Mapbox-iOS-SDK` — we would no longer have a thing named `-symbols`.)
1. Add a new `-stripped` build flavor with the explicit mandate of staying under 100 MB/file, distributed via side channels (e.g., manual feed URLs).

## Background on build flavors
We ship our iOS Maps SDK as a precompiled binary framework, which means that developers end up downloading ~100 MB up front (and storing a couple times that amount on disk, once uncompressed). While this raw, unintegrated file size has [very little bearing](https://www.mapbox.com/help/ios-framework-size/) on an app’s ultimate App Store size, in order to reduce the _apparent_ size and to help developers who put these files wholesale under source control, we’ve historically shipped two build flavors of dynamic framework:

- One with symbols stripped out into an accompanying dSYM file — this has been the default recommendation and what we specify in [our podspec](https://github.com/CocoaPods/Specs/blob/master/Specs/a/5/9/Mapbox-iOS-SDK/4.6.0/Mapbox-iOS-SDK.podspec.json).
- One that does not strip symbols — this has been an alternative build with the `-symbols` suffix, which has been available via [custom podspec](https://github.com/mapbox/mapbox-gl-native/blob/ios-v4.6.0/platform/ios/Mapbox-iOS-SDK-symbols.podspec).

## Issues with stripped binary distribution
Shipping a stripped binary by default has been a compromised approach and offers an inferior debug experience, as these builds necessarily have less debug information, spread out into more difficult to manage files. If we include more debug information in our default build (as in https://github.com/mapbox/mapbox-gl-native/pull/12672), we can no longer support the important (though minority) use case of committing everything to source control.

The stripping approach has led to a variety of issues, primarily having to do with crash logs not being able to be properly symbolicated (e.g., https://github.com/mapbox/mapbox-gl-native/issues/8463). “Binary distribution + separate symbols” is not well supported by dependency managers, fastlane, or Xcode, and it typically requires that developers [make a special effort](https://github.com/mapbox/mapbox-gl-native/pull/12257#issuecomment-401209457) to upload symbol files (such as dSYMs and bcsymbolmaps), otherwise crash logs end up being garbled or misleading.

/cc @mapbox/maps-ios @kkaefer @1ec5

### PR DESCRIPTION
**Pull request by friedbunny**
- Adds the `-stripped` build flavor — includes fewer debug symbols and strips existing ones from the framework binary. Fixes #13395.
  - Fixes #12947. For developers with file size limitations, this keeps dSYM and framework sizes under 100 MB.
- Removes the `-symbols` build flavor. The plain build now features full debugging information with only a mild download size increase (~2 MB, judging from the [4.3.0-alpha.3 release](
https://github.com/mapbox/mapbox-gl-native/releases/tag/ios-v4.7.0-alpha.3)), obviating `-symbols`.
- Adds the `RelWithDebInfo` build configuration to the iOS Xcode project — this is used for the plain build, is not stripped, and includes debug symbols for all binaries.
  - For local profiling of the SDK, `RelWithDebInfo` is the release configuration you should use.
  - Also sets `-Os` for darwin builds (#8253), while keeping `-O3` for node builds (#9154).
- Sets `GCC_GENERATE_DEBUGGING_SYMBOLS=NO` for core GL Xcode `Release` builds, while keeping it enabled for all other build configurations (#12672).
- This PR doesn’t improve our crash symbolication issues (#8463), unfortunately — so long as we ship a binary framework, we’re going to have to include a dSYM/bcsymbolmaps and make sure they’re properly submitted with apps.

/cc @mapbox/maps-ios @kkaefer

### CODE REVIEW

**Comment by friedbunny** [comment]
@julianrex @fabian-guerra I’d like to get this PR into tomorrow’s 4.3.0 pre-release, as that’s the primary way we’ll get feedback on how well these build changes actually function (and I’m on vacation for the rest of this release cycle, starting next week 😁).

File: platform/ios/CHANGELOG.md [hunk 1]
```diff
@@ -20,6 +20,7 @@ Mapbox welcomes participation and contributions from everyone. Please read [CONT
 
 ### Packaging
 
+* Added the `Mapbox-iOS-SDK-stripped` build flavor, featuring fewer debug symbols. Removed the `Mapbox-iOS-SDK-symbols` build. ([#13504](https://github.com/mapbox/mapbox-gl-native/pull/13504))
```
Comment (fabian-guerra) [inline platform/ios/CHANGELOG.md]: shouldn't we mention that the "plain" build now features full debugging information?

**Comment by friedbunny** [comment]
> Do we have a guide?

Not currently, but there’s an open documentation ticket internally asking for one.

File: platform/ios/CHANGELOG.md (hunk 1 above)
Comment (friedbunny) [reply platform/ios/CHANGELOG.md]: How about...

> Added the `Mapbox-iOS-SDK-stripped` build flavor, featuring fewer debug symbols. Regular framework binaries are no longer stripped of debug symbols and the `Mapbox-iOS-SDK-symbols` build has been retired.

Comment (fabian-guerra) [reply platform/ios/CHANGELOG.md]: This sounds good to me.

**Comment by friedbunny** [comment]
Here’s a raw file size breakdown of the regular and stripped distributions after these changes:
```
129M	../mapbox-ios-sdk-4.7.0-alpha.3-dynamic.zip
4.7M	dynamic/12C6CDF6-CAD1-3D08-AFD5-3FA80DD517CB.bcsymbolmap
4.7M	dynamic/AA83F8DD-B4AB-3B33-8FDF-295C7638DA4E.bcsymbolmap
 40K	dynamic/LICENSE.md
4.0K	dynamic/Mapbox-iOS-SDK.podspec
 91M	dynamic/Mapbox.framework
341M	dynamic/Mapbox.framework.dSYM
```
```
 47M	../mapbox-ios-sdk-4.7.0-alpha.3-stripped-dynamic.zip
1.6M	dynamic/3245D53F-B290-3631-B7B3-227E2C5D9CB4.bcsymbolmap
1.6M	dynamic/3C16954B-869B-3A04-BB31-D7DF0A2BF500.bcsymbolmap
 40K	dynamic/LICENSE.md
4.0K	dynamic/Mapbox-iOS-SDK-stripped.podspec
 54M	dynamic/Mapbox.framework
 36M	dynamic/Mapbox.framework.dSYM
```

**Pull request merged**